File: app/src/main/java/fr/mpau/models/Timer.java
package fr.mpau.models;

import java.util.List;

/**
 * Classe modèle métier d'un timer (timer = workday + workperiods)
 * <p>
 * Author: Jonathan B.
 * Created: 11/02/2018
 */

public class Timer {

    /**
     * Attributs
     */
    private int timerId;
    private WorkDay timerWorkday;
    private List<WorkPeriod> timerWorkperiodsList;

    /**
     * Constructeurs
     */
    public Timer(int timerId, WorkDay timerWorkday, List<WorkPeriod> timerWorkperiodsList) {
        this.timerId = timerId;
        this.timerWorkday = timerWorkday;
        this.timerWorkperiodsList = timerWorkperiodsList;
    }

    /**
     * Getters
     */
    public int getTimerId() {
        return timerId;
    }

    public WorkDay getTimerWorkday() {
        return timerWorkday;
    }

    public List<WorkPeriod> getTimerWorkperiodsList() {
        return timerWorkperiodsList;
    }
}

File: src/main/java/fr/mpau/bean/Timer.kt
package fr.mpau.bean

/**
 * Classe modèle métier d'un timer (timer = workday + workperiods)
 *
 * Author: Jonathan
 * Created: 11/02/2018
 * Last Updated: 24/01/2025
 */
data class Timer(

    /**
     * Attributs
     */

    val id: Int,
    val workday: WorkDay,
    val workperiodsList: List<WorkPeriod>

)
